### PR TITLE
Nerd Font data and extractor

### LINKED ISSUE #54: Font Awesome Icons?
Hello! Thanks for this awesome tool! I was thinking it would be pretty useful to integrate Font Awesome icons as a data set. Is that something someone has done? If not, how much work would it be to format them for use here? I'm assuming it just comes down to doing a [custom character file](https://github.com/fdw/rofimoji#custom-character-files), so I'll look into that unless someone has already done it.

### PR DESCRIPTION
**Pull request by katacarbix**
I've written an extractor for [Nerd Fonts](https://github.com/ryanoasis/nerd-fonts/) so users can easily insert icons from various icon sets into their projects. Closes #54 (sort of).

### How it works
1. Download the cheat sheet page: https://www.nerdfonts.com/cheat-sheet
2. Parse the data with beautifulsoup
3. Generate list of icon/classname pairs
4. Write to file

### Concerns
- Contrary to my [previous claim](https://github.com/fdw/rofimoji/issues/54#issuecomment-748764271), Font Awesome codepoints *do* appear to be the same in Nerd Fonts and their source font. However some of the other icon sets are moved to avoid conflicts within the font, meaning if someone wants to use the source icon font rather than a Nerd Font this data set may not work for them.
  - Nerd Fonts also don't include any of FA's pro icons, for obvious reasons. But also, NF has ~736 FA icons whereas FA 5 free has 1,609. Might be using an older version.
- Icon fonts aren't Unicode standard (they reside in private use blocks) and may be confusing to people who see them when they specify `--files all`. I think one way to work around this is to have a `non-standard/` directory inside `data/` which is not indexed by `--files all` but it is when specifying a filename directly. This proposed solution can decrease clutter for when others decide to contribute data sets like this one.

### CODE REVIEW

**Comment by polyzen** [comment]
> But also, NF has ~736 FA icons whereas FA 5 free has 1,609. Might be using an older version.

https://github.com/ryanoasis/nerd-fonts/issues/221

File: extractors/nerdextractor.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+    def fetch_icons(self: 'NerdExtractor'):
+        print('Downloading list of Nerd Font icons')
+
+        response = requests.get(
+            'https://www.nerdfonts.com/cheat-sheet',
+            timeout=60
+        )
+
+        characters = BeautifulSoup(response.content.decode(response.encoding), 'html.parser').find(id='glyphCheatSheet').find_all(class_='column')
```
Comment (fdw) [inline extractors/nerdextractor.py:22]: I think you can use `response.text` instead of manually decoding it.

File: extractors/nerdextractor.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+    def write_to_file(self: 'NerdExtractor'):
+        if len(self.icons) == 0:
+            return
+
+        with Path("../picker/data/nerd_font.csv").open('w') as symbol_file:
+            for icon in self.icons:
+                symbol_file.write(f'{icon.icon} {html.escape(icon.name.lower())}\n')
+
+        symbol_file.close()
```
Comment (fdw) [inline extractors/nerdextractor.py:37]: When you're using `with`, I think you can skip the explicit `close`.

**Review by fdw — CHANGES_REQUESTED** [review]
Looks good so far, thanks.

File: extractors/nerdextractor.py (hunk 1 above)
Comment (katacarbix) [reply extractors/nerdextractor.py:22]: That works too, I was just reusing code from another extractor.

**Comment by fdw** [comment]
Looks good, thank you!

**Comment by polyzen** [comment]
For some reason a large number of the nf-mdi icons appear on the right-hand side of rofi.

**Comment by fdw** [comment]
I think that is because the code point used for these symbols is reserved/used in a RTL language. I don't think there's a better solution than #62. Or do you have an idea?

**Comment by polyzen** [comment]
Oh, I see. Perhaps the extractor should automatically apply the left-to-right mark to the affected characters in nerd_fonts.csv.

**Pull request merged**